core/domain/user_services.py

```diff
@@ -1780,9 +1782,12 @@ def get_human_readable_user_ids(user_ids):
     usernames = []
     for ind, user_settings in enumerate(users_settings):
         if user_settings is None:
-            logging.error('User id %s not known in list of user_ids %s' % (
-                user_ids[ind], user_ids))
-            raise Exception('User not found.')
+            if strict:
+                logging.error('User id %s not known in list of user_ids %s' % (
+                    user_ids[ind], user_ids))
+                raise Exception('User not found.')
+            else:
+                usernames.append('[User being deleted]')
```

Comment:
"[User being deleted]" should be a named constant.

Reply:
(Doing this in a follow-up PR is fine if no other changes are requested, but please file an issue in that case to track, and link it from here.)

Reply:
Also -- could you please confirm that you've audited all the models and this sort of issue doesn't arise for any others?

Reply:
Yep, I have gone from the other side now and check pages and places where we can expect user that is being deleted and modified some code in user_services accordingly.

Reply:
And also moved this into a constant.